source/common/grpc/json_transcoder_filter.cc

```diff
@@ -217,7 +223,9 @@ Http::FilterHeadersStatus JsonTranscoderFilter::decodeHeaders(Http::HeaderMap& h
   headers.insertMethod().value().setReference(Http::Headers::get().MethodValues.Post);
   headers.insertTE().value().setReference(Http::Headers::get().TEValues.Trailers);
 
-  decoder_callbacks_->clearRouteCache();
+  if (!config_.match_incoming_request_route()) {
```

Comment:
Call decoder_callbacks_->route() in before the header conversion if `match_incoming_request_route` is set, to make sure we're not depending on the behavior that HTTP conn manager pre-calculate route. Since route is cached, this call won't impact performance.

Reply:
@ilackarms friendly reminder.